### DISCUSSION TITLE
关于AGPL-3.0许可问题

### DISCUSSION THREAD

**Discussion by 123777tw** [lifegpc/bili#76] [Q&A] [answered]
不是非常明白AGPL-3.0的具体内容。比如，如果我使用此项目的软件下载了视频并发布，是否需要注明使用此项目的软件下载？如果使用此项目的软件下载的视频进行视频剪辑等后期制作，是否还要将剪辑项目开源？

**Comment by lifegpc** [comment, marked as answer]
不需要。

**Reply by 123777tw** [reply]
感谢

**Comment by lifegpc** [comment]
AGPL的相同许可证要求是分发软件是必须要使用AGPL。
比如说A修改了我的软件，如果他自己使用，则对他没有要求（通过网络给他人使用除外）。如果他需要发布他修改过的软件，需要使用AGPL许可证发布他修改过的软件。
这个许可证对使用软件的人没有任何要求。

**Reply by 123777tw** [reply]
我个人理解，您的这些描述似乎是GPL-3.0的内容。据我在互联网上的了解，AGPL是一个非常”可怕的“协议，它会通过实际产生的服务进行协议传染——也就是我在问题中提到的顾虑，如果我们理解，使用此项目下载的视频属于此项目的实际服务内容，则使用此内容的所有项目都需要遵循AGPL。

**Reply by lifegpc** [reply]
AGPL只是加了一个网络使用也必须开源。这个指的是那个网站的源代码开源。

**Reply by lifegpc** [reply]
AGPL与GPL基本没啥区别。就只是在GPL对分发的定义的基础上增加了网络使用也算分发。这是软件的分发问题。跟使用时输出的数据没有任何关系。（

**Reply by lifegpc** [reply]
我建议你可以去看看 https://gnu.org

**Reply by 123777tw** [reply]
因为不擅长英语。非常感谢。

**Reply by lifegpc** [reply]
你提到的协议传染可能会对输出内容造成传染的是编译器（就是GCC）。因为编译器输出的内容就是二进制程序。在这种特例中，GPL许可的代码（GCC的源代码）必然会以二进制的形式静态编译到输出的二进制程序中（GPL的传染就是指以GPL许可的代码被静态编译到别的程序中，这会导致别的程序必须使用GPL），这就会导致传染。这个问题显然是被写GPL的作者关注到的。因此(A)GPL中专门处理了这个例外，使得GPL不会传染给别的程序。别的情况是一定不会传染的。
